gobblin-core/src/main/java/gobblin/source/extractor/filebased/FileBasedExtractor.java

```diff
@@ -59,7 +59,8 @@
 
   private Iterator<D> currentFileItr;
   private String currentFile;
-  private boolean readRecordStart;
+  private boolean readRecordStart = false;
```

Comment:
I don't think you need this variable. If closeCurrentFile() sets currentFile to null, then you can use null != currentFile instead of readRecordStart.